### DISCUSSION TITLE
Request for simple code editor with syntax highlight

### DISCUSSION THREAD

**Discussion by fengye87** [mantinedev/mantine#5513] [Feature requests]
I wish there's a generic code input/textarea component in Mantine, which support code highlight and is as easy to use as textarea.

There's [JsonInput](https://mantine.dev/core/json-input/) component, but it's just for json and lacks syntax highlight. Also, it's possible to handcraft something alike based on rich text editor, but that's not an easy journey.

**Comment by mattkrins** [comment]
+1 This would be very handy. I've used Mantine in multiple projects already which need a code-highlighted input.

Mantine already supports https://mantine.dev/x/code-highlight/, so perhaps this library could be combined with textarea/jsoninput, if not a base component, but as an example?

**Comment by rtivital** [comment]
This is outside of Mantine scope, this feature is not planned to be added. You can use a third-party library for this.
